File: backend/app/schemas/doctor.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, EmailStr, Field

from app.models.doctor import DoctorTier
from app.schemas.common import ORMBase


class DoctorBase(BaseModel):
    full_name: str = Field(min_length=1, max_length=160)
    specialty: str = Field(min_length=1, max_length=120)
    institution: str | None = Field(default=None, max_length=200)
    tier: DoctorTier = DoctorTier.B
    phone: str | None = Field(default=None, max_length=30)
    email: EmailStr | None = None
    address: str | None = Field(default=None, max_length=255)
    is_active: bool = True


class DoctorCreate(DoctorBase):
    pass


class DoctorUpdate(BaseModel):
    full_name: str | None = Field(default=None, min_length=1, max_length=160)
    specialty: str | None = Field(default=None, min_length=1, max_length=120)
    institution: str | None = Field(default=None, max_length=200)
    tier: DoctorTier | None = None
    phone: str | None = Field(default=None, max_length=30)
    email: EmailStr | None = None
    address: str | None = Field(default=None, max_length=255)
    is_active: bool | None = None


class DoctorRead(DoctorBase, ORMBase):
    id: UUID
    created_at: datetime
    updated_at: datetime

File: frontend/src/schemas/doctor.js
import { z } from 'zod'

/** Mirrors backend/app/schemas/doctor.py (DoctorCreate/DoctorUpdate). */
export const doctorFormSchema = z.object({
  full_name: z.string().trim().min(1, 'Full name is required.').max(160),
  specialty: z.string().trim().min(1, 'Specialty is required.').max(120),
  institution: z.string().trim().max(200).optional().or(z.literal('')),
  tier: z.enum(['A', 'B', 'C']),
  phone: z.string().trim().max(30).optional().or(z.literal('')),
  email: z.string().trim().email('Enter a valid email address.').max(160).optional().or(z.literal('')),
  address: z.string().trim().max(255).optional().or(z.literal('')),
})

export const doctorFormDefaults = {
  full_name: '',
  specialty: '',
  institution: '',
  tier: 'B',
  phone: '',
  email: '',
  address: '',
}
